### PR TITLE
Expand Raido test coverage and fix stream edge cases

### PR DESCRIPTION
**Pull request by frankvdb7**
## Summary
- Expand Common and Server test coverage across Raido behavior, infrastructure, protocol, and integration paths
- Fix empty `MemoryBufferWriter` copies
- Correct message pipe reader completion handling
- Register coverlet as a direct test dependency

## Testing
- Added unit tests covering the new edge cases and broader Raido behavior
- Not run (not requested)

### CODE REVIEW

**Review by frankvdb7 — COMMENTED** [review]
Found one correctness issue in the EOF completion handling; see the inline comment.

File: Raido.Server/RaidoMessagePipeReader.cs [hunk 1]
```diff
@@ -168,7 +168,7 @@ private bool TryCreateReadResult(ReadResult underlyingReadResult, out ReadResult
         {
             var buffer = underlyingReadResult.Buffer;
             _isCanceled = underlyingReadResult.IsCanceled;
-            _isCompleted = underlyingReadResult.IsCompleted && underlyingReadResult.Buffer.IsEmpty && !_isCanceled;
+            _isCompleted = underlyingReadResult.IsCompleted && !_isCanceled;
```
Comment (frankvdb7) [inline Raido.Server/RaidoMessagePipeReader.cs]: When a completed pipe buffer contains multiple messages, this marks the first parsed message as completed even though `_consumed` still precedes later bytes. `RaidoConnectionHandler.DispatchMessagesAsync` breaks on `result.IsCompleted`, so coalesced messages after the first are silently dropped. Surface completion only when no unread buffer/backlog remains, while still rejecting completed unparseable input.
